packages/video_player/video_player/lib/video_player.dart

```diff
@@ -760,8 +789,20 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   Future<void> _updateClosedCaptionWithFuture(
     Future<ClosedCaptionFile>? closedCaptionFile,
   ) async {
-    _closedCaptionFile = await closedCaptionFile;
-    value = value.copyWith(caption: _getCaptionAt(value.position));
+    if (closedCaptionFile != null) {
+      await closedCaptionFile.whenComplete(() async {
```

Comment:
Why are you using `whenComplete` in an async function, rather than just `await`ing the future and then using the value?

Reply:
To avoid calling the sort Everytime the events get triggered.